tls/s2n_record_write.c

```diff
@@ -89,11 +94,18 @@ int s2n_record_rounded_write_payload_size(struct s2n_connection *conn, uint16_t
     return max_fragment_size - overhead(conn);
 }
 
+/* This function returns maximum size of plaintext data to write for the payload. In the current
+  s2n implementation, it is also critical that it syncs up with from connection buffer size object */
 int s2n_record_max_write_payload_size(struct s2n_connection *conn)
 {
-    return s2n_record_rounded_write_payload_size(conn, conn->max_outgoing_fragment_length);
+    int bytes = s2n_record_rounded_write_payload_size(conn, conn->max_outgoing_fragment_length);
+    /* Record size guards here */
+    S2N_ERROR_IF(bytes > S2N_TLS_MAXIMUM_FRAGMENT_LENGTH, S2N_ERR_FRAGMENT_LENGTH_TOO_LARGE);
+    S2N_ERROR_IF(bytes <= 0, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
```

Comment:
`bytes` is about what we could write, not what we will write, right? Because trying to write `0` bytes might not be an error. I'm not super sure about TLS , but in other protocols you e.g. use that to have a method to carry an end of stream flag.

Reply:
i interpret this more as what we *could maximally* write (eg. the upper bound, not the lower bound). you could minimally write 0 bytes but it makes less sense to maximally write 0 bytes.